librz/arch/il/analysis_il.c

```diff
@@ -228,17 +228,19 @@ RZ_API bool rz_analysis_il_vm_sync_to_reg(RzAnalysisILVM *vm, RZ_NONNULL RzReg *
 	return rz_il_vm_sync_to_reg(vm->vm, vm->reg_binding, reg);
 }
 
-/**
- * Repeatedly perform steps in the VM until the condition callback returns false
- *
- * If given, this syncs the contents of \p reg into the vm.
- * Then it repeatedly disassembles an instruction at the program counter of the vm and executes it as long as cond() returns true.
- * Finally the contents are optionally synced back to \p reg.
- *
- * \return and indicator for which error occured, if any, or RZ_ANALYSIS_IL_STEP_RESULT_SUCCESS if cond() returned false
- */
-RZ_API RzAnalysisILStepResult rz_analysis_il_vm_step_while(RZ_NONNULL RzAnalysis *analysis, RZ_NONNULL RzAnalysisILVM *vm, RZ_NULLABLE RzReg *reg,
-	bool (*cond)(RzAnalysisILVM *vm, void *user), void *user) {
+static void il_events(RzILVM *vm, RzStrBuf *sb) {
+	void **it;
+	RzILEvent *evt;
+	rz_pvector_foreach (vm->events, it) {
+		evt = *it;
+		rz_il_event_stringify(evt, sb);
+		rz_strbuf_append(sb, "\n");
+	}
+}
+
+static RzAnalysisILStepResult analysis_il_vm_step_while(
+	RZ_NONNULL RzAnalysis *analysis, RZ_NONNULL RzAnalysisILVM *vm, RZ_NULLABLE RzReg *reg,
+	bool with_events, bool (*cond)(RzAnalysisILVM *vm, void *user), void *user) {
```

Comment:
`cond` is `RZ_NONNULL` and `user` is `RZ_NULLABLE`

Reply:
maybe declare as `typedef` the `bool (*cond)(RzAnalysisILVM *vm, void *user)` function